### DISCUSSION TITLE
Front-end, Back-end, DataBase, etc. selection

### DISCUSSION THREAD

**Discussion by irsaris** [uprm-inso4115-2023-2024-s2/semester-project-automated-resume-builder#4] [General]
After reading the features set for this milestone, suggest which framework you would like to work with here.

**Comment by Janiel777** [comment]
## Possible proposal

### Frontend-React
It is a framework and a javascript library to build the user interface and handle click events, keyboard inputs and maintain communication with the backend through HTTP requests.

### Backend - Express (Node.js)
It is a framework for Node.js. Express makes it easy to configure routes for APIs, handle HTTP requests and responses, integrate with databases, and other server-side operations.

### How do they interact?
React sends requests to the backend using AJAX (with the help of Axios or the Fetch API), and the backend processes these requests and responds with the requested data (for example, in JSON format).

### Database - MongoDB
For a stack that includes React on the frontend and Express with Node.js on the backend, MongoDB is a good option. It integrates well with JavaScript applications, facilitating development with a unified language across the entire technology stack (JavaScript in this case).
